internal/metastore/db/table_catalog.go

```diff
@@ -0,0 +1,850 @@
+				return err
+			}
+
+			// ii. mark deleted for collection_aliases
+			if len(collection.Aliases) > 0 {
+				err = tc.metaDomain.CollAliasDb(txCtx).MarkDeleted(tenantID, collection.CollectionID, collection.Aliases)
+				if err != nil {
+					log.Error("mark deleted for collection_aliases failed", zap.Int64("collID", collection.CollectionID), zap.Error(err))
+					return err
+				}
+			}
+
+			// iii. mark deleted for collection_channels
+			err = tc.metaDomain.CollChannelDb(txCtx).MarkDeleted(tenantID, collection.CollectionID)
+			if err != nil {
+				log.Error("mark deleted for collection_channels failed", zap.Int64("collID", collection.CollectionID), zap.Error(err))
+				return err
+			}
+
+			// iv. marked deleted for field_schemas
+			err = tc.metaDomain.FieldDb(txCtx).MarkDeleted(tenantID, collection.CollectionID)
+			if err != nil {
+				log.Error("mark deleted for field_schemas failed", zap.Int64("collID", collection.CollectionID), zap.Error(err))
+				return err
+			}
+
+			// v. marked deleted for partitions
+			err = tc.metaDomain.PartitionDb(txCtx).MarkDeleted(tenantID, collection.CollectionID)
+			if err != nil {
+				log.Error("mark deleted for partitions failed", zap.Int64("collID", collection.CollectionID), zap.Error(err))
+				return err
+			}
+		}
+
+		// 6. mark deleted for indexes
+		err := tc.metaDomain.IndexDb(txCtx).MarkDeletedByCollID(tenantID, collection.CollectionID)
+		if err != nil {
+			log.Error("mark deleted for indexes failed", zap.Error(err))
+			return err
+		}
+
+		// 7. mark deleted for segment_indexes
+		err = tc.metaDomain.SegmentIndexDb(txCtx).MarkDeletedByCollID(tenantID, collection.CollectionID)
+		if err != nil {
+			log.Error("mark deleted for segment_indexes failed", zap.Error(err))
+			return err
+		}
+
+		return err
+	})
+}
+
+func (tc *Catalog) CreatePartition(ctx context.Context, coll *model.Collection, ts typeutil.Timestamp) error {
+	tenantID := contextutil.TenantID(ctx)
+
+	partition := coll.Partitions[0]
+	p := &dbmodel.Partition{
+		TenantID:                  tenantID,
+		PartitionID:               partition.PartitionID,
+		PartitionName:             partition.PartitionName,
+		PartitionCreatedTimestamp: partition.PartitionCreatedTimestamp,
+		CollectionID:              coll.CollectionID,
+		Ts:                        ts,
+	}
+	err := tc.metaDomain.PartitionDb(ctx).Insert([]*dbmodel.Partition{p})
+	if err != nil {
+		log.Error("insert partitions failed", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func (tc *Catalog) DropPartition(ctx context.Context, collection *model.Collection, partitionID typeutil.UniqueID, ts typeutil.Timestamp) error {
+	tenantID := contextutil.TenantID(ctx)
+	if ts > 0 {
+		p := &dbmodel.Partition{
+			TenantID:     tenantID,
+			PartitionID:  partitionID,
+			CollectionID: collection.CollectionID,
+			Ts:           ts,
+			IsDeleted:    true,
+		}
+		err := tc.metaDomain.PartitionDb(ctx).Insert([]*dbmodel.Partition{p})
```

Comment:
Same as above, `Insert` here cause `DropParttion` not idempotent, which is not consistent with kv catalog.